.github/workflows/pypi-package.yml

```diff
@@ -0,0 +1,72 @@
+---
+name: Build and upload package
+
+on:
+  push:
+    branches: [main]
+    tags: ["*"]
+  release:
+    types:
+      - published
+  workflow_dispatch:
+
+permissions:
+  contents: read
+  id-token: write
+
+jobs:
+  build-package:
+    name: Build & check package
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        submodules: recursive
+        fetch-depth: 0
+    - name: Set up Python 3.12
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.12
+    - run: pip install --upgrade build twine
+    - name: Build sdist and wheel
+      run: python -m build
+    - name: Check package
+      run: twine check dist/*
+    - name: Upload sdist and wheel artifacts
+      uses: actions/upload-artifact@v4
+      with:
+        name: dist
+        path: dist/
+
+  auto-release-test-pypi:
+    runs-on: ubuntu-latest
+    environment: "Test-release"
```

Comment:
We don't need an environment for test pushes. It either does nothing or spams maintainers for approval on every push to `main`.

```suggestion
```

Reply:
Thanks, indeed, it makes sense. I wasn't sure if we should also restrict the test releases.